pyiron/atomistics/master/elastic.py

```diff
@@ -0,0 +1,223 @@
+        coeff = np.triu(np.ones((6,6))).flatten()
+        coeff[coeff!=0] *= reg.coef_[1:]*eV_div_A3_to_GPa
+        coeff = coeff.reshape(6,6)
+        coeff = 0.5*(coeff+coeff.T)
+    else:
+        raise ValueError('Problem with fitting data')
+    return coeff
+
+def calc_elastic_constants(elastic_tensor):
+    """
+    Calculate elastic constants from the elastic tensor.
+
+    For anistropic material (i.e. zener_ratio!=1), the values may or may not make
+    sense -> don't trust the results straightforwardly
+
+    Args:
+        elastic_tensor (numpy.ndarray): 6x6 tensor
+    """
+    output = {}
+    output['elastic_tensor'] = elastic_tensor
+    output['lame_coefficient'] = np.mean(elastic_tensor[:3, :3].diagonal())
+    output['shear_modulus'] = np.mean(elastic_tensor[3:, 3:].diagonal())
+    output['bulk_modulus'] = np.mean(elastic_tensor[:3,:3])
+    output['poissons_ratio'] = 0.5/(1+output['shear_modulus']/output['lame_coefficient'])
+    output['youngs_modulus'] = 1/np.mean(np.linalg.inv(elastic_tensor[:3,:3]).diagonal())
+    output['poissons_ratio'] = -output['youngs_modulus']*np.sum(np.linalg.inv(elastic_tensor[:3,:3]))/6+0.5
+    output['zener_ratio'] = 12*np.mean(elastic_tensor[3:,3:].diagonal())/(3*np.trace(elastic_tensor[:3,:3])-np.sum(elastic_tensor[:3,:3]))
+    return output
+
+class ElasticJobGenerator(JobGenerator):
+    @property
+    def parameter_list(self):
+        """
+
+        Returns:
+            (list)
+        """
+        parameter_lst = []
+        for ii, epsilon in enumerate(self._job.input['strain_matrices']):
+            basis = self._job.ref_job.structure.copy()
+            basis.apply_strain(np.array(epsilon))
+            parameter_lst.append([ii, basis])
+        return parameter_lst
+
+    def job_name(self, parameter):
+        return "{}_{}".format(self._job.job_name, parameter[0]).replace(".", "_")
+
+    @staticmethod
+    def modify_job(job, parameter):
+        job.structure = parameter[1]
+        return job
+
+# ToDo: not all abstract methods implemented
+class ElasticTensor(AtomisticParallelMaster):
+    """
+    Class to calculate the elastic tensor and isotropic elastic constants
+
+    Example:
+
+    >>> job = pr.create_job('SomeAtomisticJob', 'atomistic')
+    >>> job.structure = pr.create_structure('Fe', 'bcc', 2.83)
+    >>> elastic = job.create_job('ElasticTensor', 'elastic')
+    >>> elastic.run()
+
+    This class is still under construction and there's no guarantee that it does
+    what it looks like it does.
+    """
+    def __init__(self, project, job_name="elastic"):
+        """
+
+        Args:
+            project:
+            job_name:
+        """
+        super().__init__(project, job_name)
+        self.__name__ = "ElasticTensor"
+        self.__version__ = "0.1.0"
+
+        # print ("h5_path: ", self.project_hdf5._h5_path)
+
+        # define default input
+        self.input["min_num_measurements"] = (11, "minimum number of samples to be taken")
```

Comment:
`self.input` should be an `InputList`. This doesn't allow for giving the docstring in the param definition, but you can look over in the PR Marvin mentioned for my proposed solution to that. Briefly: at the class level in addition to `Attributes` I propose to add `Input` and `Output` lists giving the keys and values of these `InputList`s (-->`IOList`` one day)

Reply:
If you do this you'll probably need to add `self.input.to_hdf(); self.output.to_hdf()` to the `to_hdf` method (and similar for `from_hdf`) as the parent job classes should but don't yet handle `InputList` being used this way.

Reply:
> `self.input` should be an `InputList`.

DEFINITELY. The problem is right now it's derived from `AtomisticParallelMaster` (or `ParallelMaster`) and I have little control over the original input class...

Reply:
> > `self.input` should be an `InputList`.
> 
> DEFINITELY. The problem is right now it's derived from `AtomisticParallelMaster` (or `ParallelMaster`) and I have little control over the original input class...

You can overwrite the input class.

Reply:
Not sure if this is a good idea, as the `to_hdf`/`from_hdf` of the base class might expect `input` to be a `GenericParameter` object.  I haven't checked the source though.

Reply:
No, it should be fine. You'll need to explicitly handle the hdf operations for input and output here (unless @niklassiemer shifts his changes in [pyiron_base PR 50] up the inheritance tree like I hope he will 😉). Otherwise it's worked fine for me.

I still want to see this!